pyproject.toml

```diff
@@ -1,3 +1,130 @@
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft",
+    "Operating System :: Unix",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Communications",
+    "Topic :: Internet",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Software Development",
+    "Topic :: System",
+]
+
+[tool.poetry.scripts]
+aea = 'aea.cli:cli'
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.11"
+base58 = ">=1.0.3,<3.0.0"
+jsonschema = ">=3.0.0,<4.0.0"
+packaging = ">=20.3,<21.0"
+semver = ">=2.9.1,<3.0.0"
+protobuf = ">=3.19.0,<=3.19.4"
+pymultihash = "==0.8.2"
+pyyaml = ">=4.2b1,<6.0"
+requests = ">=2.22.0,<3.0.0"
+python-dotenv = ">=0.14.0,<0.18.0"
+ecdsa = ">=0.15,<0.17.0"
+
+pywin32 = { version = "==303", markers = "sys_platform == 'win32' or platform_system == 'Windows'" }
+
+# cli, test_tools extras
+click = { version = ">=7.0.0,<8.0.0", optional = true }
+
+
+[tool.poetry.extras]
+cli = ["click"]
+test_tools = ["click"]
+all=["click"]
+
+
+[tool.poetry.group.dev.dependencies]
+tox = "==3.25.1"
+py-sr25519-bindings = "==0.1.4"
+cosmpy = ">=0.5.0,<0.6.0"
+
+
+[tool.poetry.group.linting.dependencies]
+pylint = "==2.6.0"
+bandit = "==1.7.0"
+vulture = "==2.3"
+isort = "==5.7.0"
+safety = "==1.10.3"
+mypy = "==0.761"
+darglint = "==1.8.1"
+flake8 = "==3.7.9"
+flake8-bugbear = "==20.1.4"
+flake8-docstrings = "==1.6.0"
+flake8-eradicate = "==0.4.0"
+flake8-isort = "==4.0.0"
+liccheck = "==0.6.0"
+black = "==19.10b0"
+
+[tool.poetry.group.testing.dependencies]
+pytest = "==7.0.0"
+pytest-asyncio = "==0.16.0"
+pytest-cov = "==3.0.0"
+pytest-custom-exit-code = "==0.3.0"
+pytest-randomly = "==3.10.3"
+pytest-rerunfailures = "==10.2"
+docker = "==4.2.0"
+pexpect = "==4.8.0"
+sqlalchemy = "==1.4.17"  # used in one test
+mistune = "==2.0.0a4"
+requests = ">=2.22.0"
+web3 = "web3==5.12.0"
+
+[tool.poetry.group.docs.dependencies]
+mkdocs-material = "==6.2.8"
+mkdocs-mermaid-plugin = { git = "https://github.com/pugong/mkdocs-mermaid-plugin.git" }
+pydoc-markdown = "*"
+pydocstyle = "==3.0.0"
+pymdown-extensions = "==9.1"
+pygments = "==2.7.4"
+
+
+[tool.poetry.group.packages.dependencies]
+yoti = "==2.14.0"
+tensorflow = [{ version = "^2.9.2", python = "<3.11" }]
+#openapi-core = "==0.13.2"  # problematic package for poetry
+openapi-spec-validator = "==0.2.8"  # required by openapi core and best to pin it here
+gym = "==0.15.6"
+aiohttp = ">=3.7.4,<3.8"
+aioprometheus = ">=20.0.0,<21.0.0"
+numpy = ">=1.18.1"
+oef = "==0.8.1"
+defusedxml = "==0.7.1"
+scikit-image = ">=0.17.2"
+colorlog = "==4.1.0"
+temper-py = "==0.0.3"
+
+
+[tool.poetry.group.tools.dependencies]
+#benchmark
+matplotlib = ">=3.3.0,<3.4"
+
+gitpython = ">=3.1.14"
+ipfshttpclient = "==0.6.1"
+memory-profiler = "==0.57.0"
+psutil = "==5.7.0"
```

Comment:
Why these grouping generally?

Reply:
Also there are some duplicates in the groups. Is that alright?

Reply:
Let's monitor this separation of deps and decide later on the suitability of this.